src/tribler/core/components/tunnel/community/dispatcher.py

```diff
@@ -120,8 +126,13 @@ def select_circuit(self, connection, request):
                 return None
             self._logger.debug("Creating circuit for data to %s. Retrying later..", request.destination)
             self.cid_to_con[circuit.circuit_id] = connection
-            circuit.ready.add_done_callback(lambda f, c=connection.udp_connection, r=request:
-                                            self.on_socks5_udp_data(c, r) if f.result() else None)
+            circuit.ready.add_done_callback(
+                functools.partial(
+                    self._add_data_if_result,
+                    connection=connection.udp_connection,
+                    request=request
+                )
+            )
```

Comment:
Could you please consider simplifying it, something along the lines of the following:
```suggestion
 
            def done_callback(f, c=connection.udp_connection, r=request):
                return self.on_socks5_udp_data(c, r) if f.result() else None

            circuit.ready.add_done_callback(done_callback)
```

Reply:
Will do.